fastly/waf.go

```diff
@@ -0,0 +1,133 @@
+package fastly
+
+import "fmt"
+
+// SetData represents information about a configuration_set.
+type SetData struct {
+	Type string `mapstructure:"type"`
+	ID   string `mapstructure:"id"`
+}
+
+// FirewallConfigurationSet is the data inside of a configuration_set relationship.
+type FirewallConfigurationSet struct {
+	Data SetData `mapstructure:"data"`
+}
+
+// FirewallRelationships holds the relationships of a firewall object (currently just `configuration_set`).
+type FirewallRelationships struct {
+	ConfigurationSet FirewallConfigurationSet `mapstructure:"configuration_set"`
+}
+
+// FirewallAttributes holds the attributes of a firewall object.
+type FirewallAttributes struct {
+	PrefetchCondition string `mapstructure:"prefetch_condition"`
+	Response          string `mapstructure:"response"`
+	LastPush          string `mapstructure:"last_push"`
+	Version           string `mapstructure:"version"`
+}
+
+// FirewallData is the information about a firewall object.
+type FirewallData struct {
+	ID            string                `mapstructure:"id"`
+	Type          string                `mapstructure:"type"`
+	Attributes    FirewallAttributes    `mapstructure:"attributes"`
+	Relationships FirewallRelationships `mapstructure:"relationships"`
+}
+
+// FirewallLinks specifies first, last, and previous (if applicable) page links.
+type FirewallLinks struct {
+	Last     string `mapstructure:"last"`
+	First    string `mapstructure:"first"`
+	Previous string `mapstructure:"previous"`
+}
+
+// FirewallObjects represents a response from Fastly's API for listing firewall objects.
+type FirewallObjects struct {
+	// Data lists all of the firewall objects.
+	Data []FirewallData `mapstructure:"data"`
+
+	// Included lists relationships.
+	Included []FirewallData `mapstructure:"included"`
+
+	// Lists links in relation to your current page.
+	Links FirewallLinks `mapstructure:"links"`
+}
+
+// GetFirewallObjectsInput is the input needed to list firewall objects.
+type GetFirewallObjectsInput struct {
+	// Service is the ID of the service (required).
+	Service string
```

Comment:
I think all of the `*Input` structs need to declare the `form:"field,omitempty"` values